proto/asn1-pdu/asn1_universal_types_to_der.cc

```diff
@@ -67,6 +67,39 @@ void Encode(const BitString& bit_string, std::vector<uint8_t>& der) {
   }
 }
 
+void Encode(const ObjectIdentifier& object_identifier,
+            std::vector<uint8_t>& der) {
+  // Save the current size in |tag_len_pos| to place tag and length
+  // after the value is encoded.
+  const size_t tag_len_pos = der.size();
+
+  uint8_t root = object_identifier.root();
+  uint8_t small_identifier = object_identifier.small_identifier();
+  auto subidentifier = object_identifier.subidentifier();
+
+  // (X.690 (2015) 8.19.4): Only 39 subsequent values from nodes reached by X =
+  // 0 and X = 1. Therefore, use |small_identifier| for |root| 0 or 1, and when
+  // |root| is 2, use first integer in |subidentifier| to obtain
+  // potentially higher values.
+  size_t identifier = (root == 2 && !subidentifier.empty())
+                          ? (root * 40) + subidentifier[0]
+                          : (root * 40) + small_identifier;
+  der.push_back(identifier);
```

Comment:
BUG: You push back as a size_t, but that doesn't yield correct results for situations where `identifier` > 255.

The best way to make sure you've got this fixed is to use the example following 8.19.5 of X.690 (2015), where the OID `{2 999 3}` is encoded as `06 03 883703`.

That is, 80 + 999 = 1079, which then gets base128 encoded.

The easiest way to fix this is just:
```suggestion
InsertVariableIntBase128(identifier, der.size(), der);
```

Reply:
Yes, it keeps slipping my mind that we need to base 128 encode values above 255. Thanks for catching this once again.